### PR TITLE
[Dungeon] Scarlet Monastery

### PR DESCRIPTION
**Pull request by Knindzagxg**
Changes:

Min, Max & Modifier Damage for whole Dungeon.
Added flags for each boss, so they will appear with skull instead regular level elite.
Missing AI included now as well.

All checks have been done trough Classic Realms and Youtube videos.

How to test:
Apply SQL
Restart Server and delete your Cache.

Make sure to add very very basic gear towards each Dungeon.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1588337316614118000.sql [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+(4285, 0, 1, 0, 2, 0, 100, 1, 25, 50, 0, 0, 11, 11640, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Scarlet Disciple - Between 25-50% Health - Cast \'11640\' (No Repeat)'),
+(4285, 0, 2, 0, 2, 0, 100, 1, 0, 10, 0, 0, 11, 11642, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Scarlet Disciple - Between 0-10% Health - Cast \'11642\' (No Repeat)');
+
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 4283;
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 4283);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(4283, 0, 0, 0, 2, 0, 100, 1, 20, 80, 0, 0, 11, 3639, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Scarlet Sentry - Between 20-80% Health - Cast \'3639\' (No Repeat)');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337316614118000.sql:64]: newline
```suggestion
(4283, 0, 0, 0, 2, 0, 100, 1, 20, 80, 0, 0, 11, 3639, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Scarlet Sentry - Between 20-80% Health - Cast \'3639\' (No Repeat)');

```

File: data/sql/updates/pending_db_world/rev_1588337316614118000.sql [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+UPDATE `creature_template` SET `mindmg` = 143, `maxdmg` = 199, `DamageModifier` = 1.03 WHERE `entry` = 4301;
+UPDATE `creature_template` SET `mindmg` = 30, `maxdmg` = 42, `DamageModifier` = 1.03 WHERE `entry` = 6575;
+UPDATE `creature_template` SET `mindmg` = 90, `maxdmg` = 129, `DamageModifier` = 1.03 WHERE `entry` = 4300;
+UPDATE `creature_template` SET `mindmg` = 87, `maxdmg` = 125, `DamageModifier` = 1.03 WHERE `entry` = 4294;
+UPDATE `creature_template` SET `mindmg` = 121, `maxdmg` = 161, `DamageModifier` = 1.03 WHERE `entry` = 4302;
+UPDATE `creature_template` SET `mindmg` = 121, `maxdmg` = 161, `DamageModifier` = 1.03 WHERE `entry` = 4303;
+
+/* RARE */
+UPDATE `creature_template` SET `rank` = 2, `mindmg` = 174, `maxdmg` = 224, `DamageModifier` = 1.02 WHERE `entry` = 6490;
+UPDATE `creature_template` SET `rank` = 2, `mindmg` = 174, `maxdmg` = 224, `DamageModifier` = 1.02 WHERE `entry` = 6488;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337316614118000.sql:40]: Just to make less queries
```suggestion
UPDATE `creature_template` SET `rank` = 2, `mindmg` = 174, `maxdmg` = 224, `DamageModifier` = 1.02 WHERE `entry` IN (6490, 6488);
```

File: data/sql/updates/pending_db_world/rev_1588337316614118000.sql [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+UPDATE `creature_template` SET `mindmg` = 83, `maxdmg` = 118, `DamageModifier` = 1.03 WHERE `entry` = 4289;
+UPDATE `creature_template` SET `mindmg` = 108, `maxdmg` = 147, `DamageModifier` = 1.03 WHERE `entry` = 4292;
+UPDATE `creature_template` SET `mindmg` = 184, `maxdmg` = 255, `DamageModifier` = 1.03 WHERE `entry` = 4290;
+UPDATE `creature_template` SET `mindmg` = 115, `maxdmg` = 155, `DamageModifier` = 1.03 WHERE `entry` = 4295;
+UPDATE `creature_template` SET `mindmg` = 143, `maxdmg` = 199, `DamageModifier` = 1.03 WHERE `entry` = 4301;
+UPDATE `creature_template` SET `mindmg` = 30, `maxdmg` = 42, `DamageModifier` = 1.03 WHERE `entry` = 6575;
+UPDATE `creature_template` SET `mindmg` = 90, `maxdmg` = 129, `DamageModifier` = 1.03 WHERE `entry` = 4300;
+UPDATE `creature_template` SET `mindmg` = 87, `maxdmg` = 125, `DamageModifier` = 1.03 WHERE `entry` = 4294;
+UPDATE `creature_template` SET `mindmg` = 121, `maxdmg` = 161, `DamageModifier` = 1.03 WHERE `entry` = 4302;
+UPDATE `creature_template` SET `mindmg` = 121, `maxdmg` = 161, `DamageModifier` = 1.03 WHERE `entry` = 4303;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337316614118000.sql:36]:
```suggestion
UPDATE `creature_template` SET `mindmg` = 121, `maxdmg` = 161, `DamageModifier` = 1.03 WHERE `entry` IN (4302, 4303);
```

File: data/sql/updates/pending_db_world/rev_1588337316614118000.sql [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+/* RARE */
+UPDATE `creature_template` SET `rank` = 2, `mindmg` = 174, `maxdmg` = 224, `DamageModifier` = 1.02 WHERE `entry` = 6490;
+UPDATE `creature_template` SET `rank` = 2, `mindmg` = 174, `maxdmg` = 224, `DamageModifier` = 1.02 WHERE `entry` = 6488;
+UPDATE `creature_template` SET `rank` = 2, `mindmg` = 166, `maxdmg` = 215, `DamageModifier` = 1.02 WHERE `entry` = 6489;
+
+/* BOSS */
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` = 3983;
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` = 4543;
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` = 3974;
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` = 6487;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337316614118000.sql:47]:
```suggestion
UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` IN (3983, 4543, 3974, 6487);
```

File: data/sql/updates/pending_db_world/rev_1588337316614118000.sql [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 215, `maxdmg` = 276, `DamageModifier` = 1.01 WHERE `entry` = 3975;
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 229, `maxdmg` = 295, `DamageModifier` = 1.01 WHERE `entry` = 4542;
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 269, `maxdmg` = 346, `DamageModifier` = 1.01 WHERE `entry` = 3976;
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 269, `maxdmg` = 346, `DamageModifier` = 1.01 WHERE `entry` = 3977;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337316614118000.sql:51]:
```suggestion
UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 269, `maxdmg` = 346, `DamageModifier` = 1.01 WHERE `entry` IN (3976, 3977);
```

File: data/sql/updates/pending_db_world/rev_1588337316614118000.sql [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+/*
+ * Dungeon: Scarlet Monastery
+ * Update by Knindza | <www.azerothcore.org>
+ * Copyright (C) <www.shadowburn.net> & <www.lichbane.com>
+*/
+
+/* REGULAR */ 
+UPDATE `creature_template` SET `mindmg` = 112, `maxdmg` = 156, `DamageModifier` = 1.03 WHERE `entry` = 4823;
+UPDATE `creature_template` SET `mindmg` = 92, `maxdmg` = 124, `DamageModifier` = 1.03 WHERE `entry` = 6547;
+UPDATE `creature_template` SET `mindmg` = 112, `maxdmg` = 156, `DamageModifier` = 1.03 WHERE `entry` = 4306;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337316614118000.sql:12]:
```suggestion
UPDATE `creature_template` SET `mindmg` = 112, `maxdmg` = 156, `DamageModifier` = 1.03 WHERE `entry` IN (4823, 4306);
UPDATE `creature_template` SET `mindmg` = 92, `maxdmg` = 124, `DamageModifier` = 1.03 WHERE `entry` = 6547;
```

File: data/sql/updates/pending_db_world/rev_1588337316614118000.sql [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+UPDATE `creature_template` SET `mindmg` = 116, `maxdmg` = 156, `DamageModifier` = 1.03 WHERE `entry` = 6427;
+UPDATE `creature_template` SET `mindmg` = 146, `maxdmg` = 189, `DamageModifier` = 1.03 WHERE `entry` = 6493;
+UPDATE `creature_template` SET `mindmg` = 116, `maxdmg` = 156, `DamageModifier` = 1.03 WHERE `entry` = 6426;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337316614118000.sql:15]:
```suggestion
UPDATE `creature_template` SET `mindmg` = 116, `maxdmg` = 156, `DamageModifier` = 1.03 WHERE `entry` IN (6427, 6426);
UPDATE `creature_template` SET `mindmg` = 146, `maxdmg` = 189, `DamageModifier` = 1.03 WHERE `entry` = 6493;
```

File: data/sql/updates/pending_db_world/rev_1588337316614118000.sql [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+UPDATE `creature_template` SET `mindmg` = 112, `maxdmg` = 156, `DamageModifier` = 1.03 WHERE `entry` = 4308;
+UPDATE `creature_template` SET `mindmg` = 103, `maxdmg` = 140, `DamageModifier` = 1.03 WHERE `entry` = 4287;
+UPDATE `creature_template` SET `mindmg` = 103, `maxdmg` = 140, `DamageModifier` = 1.03 WHERE `entry` = 4296;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337316614118000.sql:18]:
```suggestion
UPDATE `creature_template` SET `mindmg` = 103, `maxdmg` = 140, `DamageModifier` = 1.03 WHERE `entry` IN (4287, 4296);
```

File: data/sql/updates/pending_db_world/rev_1588337316614118000.sql [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+UPDATE `creature_template` SET `mindmg` = 124, `maxdmg` = 171, `minrangedmg` = 89, `maxrangedmg` = 136, `DamageModifier` = 1.03 WHERE `entry` = 4288;
+UPDATE `creature_template` SET `mindmg` = 93, `maxdmg` = 125, `DamageModifier` = 1.03 WHERE `entry` = 4304;
+UPDATE `creature_template` SET `mindmg` = 106, `maxdmg` = 140, `DamageModifier` = 1.03 WHERE `entry` = 4299;
+UPDATE `creature_template` SET `mindmg` = 106, `maxdmg` = 140, `DamageModifier` = 1.03 WHERE `entry` = 4291;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337316614118000.sql:22]:
```suggestion
UPDATE `creature_template` SET `mindmg` = 106, `maxdmg` = 140, `DamageModifier` = 1.03 WHERE `entry` IN (4299, 4291);
```

File: data/sql/updates/pending_db_world/rev_1588337316614118000.sql [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+/* RARE */
+UPDATE `creature_template` SET `rank` = 2, `mindmg` = 174, `maxdmg` = 224, `DamageModifier` = 1.02 WHERE `entry` IN (6490, 6488);
+UPDATE `creature_template` SET `rank` = 2, `mindmg` = 166, `maxdmg` = 215, `DamageModifier` = 1.02 WHERE `entry` = 6489;
+
+/* BOSS */
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` IN (3983, 4543, 3974, 6487);
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 215, `maxdmg` = 276, `DamageModifier` = 1.01 WHERE `entry` = 3975;
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 229, `maxdmg` = 295, `DamageModifier` = 1.01 WHERE `entry` = 4542;
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 269, `maxdmg` = 346, `DamageModifier` = 1.01 WHERE `entry` IN (3976, 3977);
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337316614118000.sql:41]:
```suggestion
UPDATE `creature_template` SET `type_flags`=`type_flags`|4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` IN (3983, 4543, 3974, 6487);
UPDATE `creature_template` SET `type_flags`=`type_flags`|4, `mindmg` = 215, `maxdmg` = 276, `DamageModifier` = 1.01 WHERE `entry` = 3975;
UPDATE `creature_template` SET `type_flags`=`type_flags`|4, `mindmg` = 229, `maxdmg` = 295, `DamageModifier` = 1.01 WHERE `entry` = 4542;
UPDATE `creature_template` SET `type_flags`=`type_flags`|4, `mindmg` = 269, `maxdmg` = 346, `DamageModifier` = 1.01 WHERE `entry` IN (3976, 3977);
```

**Comment by Geekena** [comment]
in first one on the left side i saw only 1 rare
everything else is ok

**Comment by Knindzagxg** [comment]
That's normal mate.
There is a chance to have 1 up (2 different Rares inside) and sometimes you won't have any at all.

**Comment by Kitzunu** [comment]
Ty for the PR!

**Pull request merged**